drivers/auxdisplay/auxdisplay_serlcd.c

```diff
@@ -0,0 +1,436 @@
+			rc = auxdisplay_serlcd_send_command(dev, command);
+			if (!rc) {
+				return rc;
+			}
+			auxdisplay_serlcd_advance_current_position(dev);
+		} else if (character == SERLCD_BEGIN_COMMAND ||
+			   character == SERLCD_BEGIN_SPECIAL_COMMAND) {
+			/*
+			 * skip these characters in text, as they have a special meaning, if
+			 * required a custom character can be used as replacement
+			 */
+			continue;
+		} else {
+			rc = i2c_write_dt(&config->bus, text, len);
+			if (!rc) {
+				return rc;
+			}
+			auxdisplay_serlcd_advance_current_position(dev);
+		}
+	}
+
+	return rc;
+}
+
+static int auxdisplay_serlcd_init(const struct device *dev)
+{
+	const struct auxdisplay_serlcd_config *config = dev->config;
+	struct auxdisplay_serlcd_data *data = dev->data;
+
+	/*
+	 * Initialize our data structure
+	 */
+	data->power = true;
+
+	if (!device_is_ready(config->bus.bus)) {
+		return -ENODEV;
+	}
+
+	auxdisplay_serlcd_clear(dev);
+
+	return 0;
+}
+
+static const struct auxdisplay_driver_api auxdisplay_serlcd_auxdisplay_api = {
+	.display_on = auxdisplay_serlcd_display_on,
+	.display_off = auxdisplay_serlcd_display_off,
+	.cursor_set_enabled = auxdisplay_serlcd_cursor_set_enabled,
+	.position_blinking_set_enabled = auxdisplay_serlcd_position_blinking_set_enabled,
+	.cursor_position_set = auxdisplay_serlcd_cursor_position_set,
+	.cursor_position_get = auxdisplay_serlcd_cursor_position_get,
+	.capabilities_get = auxdisplay_serlcd_capabilities_get,
+	.clear = auxdisplay_serlcd_clear,
+	.custom_character_set = auxdisplay_serlcd_custom_character_set,
+	.write = auxdisplay_serlcd_write,
+};
+
+#define AUXDISPLAY_SERLCD_INST(inst)                                                               \
+	static const struct auxdisplay_serlcd_config auxdisplay_serlcd_config_##inst = {           \
+		.capabilities = {                                                                  \
+				.columns = DT_INST_PROP(inst, columns),                            \
+				.rows = DT_INST_PROP(inst, rows),                                  \
+				.mode = 0,                                                         \
+				.brightness.minimum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,              \
+				.brightness.maximum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,              \
+				.backlight.minimum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,               \
+				.backlight.maximum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,               \
+				.custom_characters = SERLCD_CUSTOM_CHAR_MAX_COUNT,                 \
+				.custom_character_width = SERLCD_CUSTOM_CHAR_WIDTH,                \
+				.custom_character_height = SERLCD_CUSTOM_CHAR_HEIGHT,              \
+			},                                                                         \
+		.bus = I2C_DT_SPEC_INST_GET(inst)};                                                \
+                                                                                                   \
+	static struct auxdisplay_serlcd_data auxdisplay_serlcd_data_##inst = {};                   \
```

Comment:
just

`static struct auxdisplay_serlcd_data auxdisplay_serlcd_data_##inst;                   \` (drop the ` = {}`)

Static variables are 0 initialized automatically.

Reply:
Done